cmd/operator-sdk/run/local.go

```diff
@@ -76,65 +85,40 @@ func (c runLocalArgs) run() error {
 	return projutil.ErrUnknownOperatorType{}
 }
 
+// runGo will run the project locally for the Go Type projects
 func (c runLocalArgs) runGo() error {
-	projutil.MustInProjectRoot()
-	absProjectPath := projutil.MustGetwd()
-	projectName := filepath.Base(absProjectPath)
-	outputBinName := filepath.Join(scaffold.BuildBinDir, projectName+"-local")
-	if runtime.GOOS == "windows" {
-		outputBinName += ".exe"
-	}
-	if err := c.buildLocal(outputBinName); err != nil {
-		return fmt.Errorf("failed to build operator to run locally: %v", err)
-	}
-
-	args := []string{}
-	if c.operatorFlags != "" {
-		extraArgs := strings.Split(c.operatorFlags, " ")
-		args = append(args, extraArgs...)
+	// Build the project and generate binary that will be executed
+	binName, err := c.generateBinary()
+	if err != nil {
+		return err
 	}
-
-	var dc *exec.Cmd
-
+	// Get the args that will be used to exec the binary.
+	// Users are allowed to use the flag operator-flags to pass any value that they may wish
+	args := c.buildArgsBasedOnOperatorFlag()
+	// Build the command
+	var cmd *exec.Cmd
 	if c.enableDelve {
-		delveArgs := []string{"--listen=:2345", "--headless=true", "--api-version=2", "exec", outputBinName, "--"}
-		delveArgs = append(delveArgs, args...)
-
-		dc = exec.Command("dlv", delveArgs...)
-		log.Infof("Delve debugger enabled with args %s", delveArgs)
+		cmd = getExecCmdWithDebugger(binName, args)
 	} else {
-		dc = exec.Command(outputBinName, args...)
+		cmd = exec.Command(binName, args...)
 	}
-
+	// Kills the binary execution, if the exit signal be raised from the container
 	ch := make(chan os.Signal)
 	signal.Notify(ch, os.Interrupt, syscall.SIGTERM)
 	go func() {
 		<-ch
-		err := dc.Process.Kill()
+		err := cmd.Process.Kill()
 		if err != nil {
 			log.Fatalf("Failed to terminate the operator: (%v)", err)
 		}
 		os.Exit(0)
 	}()
-	dc.Env = os.Environ()
-	dc.Env = append(dc.Env, fmt.Sprintf("%s=%s", k8sutil.ForceRunModeEnv, k8sutil.LocalRunMode))
-	// only set env var if user explicitly specified a kubeconfig path
-	if c.kubeconfig != "" {
-		dc.Env = append(dc.Env, fmt.Sprintf("%v=%v", k8sutil.KubeConfigEnvVar, c.kubeconfig))
-	}
-	dc.Env = append(dc.Env, fmt.Sprintf("%v=%v", k8sutil.WatchNamespaceEnvVar, c.watchNamespace))
-
-	// Set the ANSIBLE_ROLES_PATH
-	if c.ansibleOperatorFlags != nil && len(c.ansibleOperatorFlags.AnsibleRolesPath) > 0 {
-		log.Info(fmt.Sprintf("set the value %v for environment variable %v.", c.ansibleOperatorFlags.AnsibleRolesPath,
-			aoflags.AnsibleRolesPathEnvVar))
-		dc.Env = append(dc.Env, fmt.Sprintf("%v=%v", aoflags.AnsibleRolesPathEnvVar, c.ansibleOperatorFlags.AnsibleRolesPath))
-	}
-
-	if err := projutil.ExecCmd(dc); err != nil {
-		return fmt.Errorf("failed to run operator locally: %v", err)
+	// Add default env vars and values informed via flags
+	c.addEnvVars(cmd)
+	// Execute the command build above to exec the binary build with the project
```

Comment:
Remove this comment. It's not adding anything more context to `ExecCmd` which is clear enough imo.
```suggestion
```